producer/callback.go

```diff
@@ -39,6 +48,59 @@ func HandleSmPolicyNotify(request *httpwrapper.Request) *httpwrapper.Response {
 	return httpwrapper.NewResponse(http.StatusNotImplemented, nil, nil)
 }
 
-// TODO: handle SM Policy Notify
+// SmPolicyNotifyProcedure TODO: handle SM Policy Notify
 func SmPolicyNotifyProcedure(supi string, notification models.PolicyDataChangeNotification) {
 }
+
+func HandleNfSubscriptionStatusNotify(request *httpwrapper.Request) *httpwrapper.Response {
+	logger.ProducerLog.Traceln("Handle NF Status Notify")
+
+	notificationData := request.Body.(models.NotificationData)
+
+	problemDetails := NfSubscriptionStatusNotifyProcedure(notificationData)
+	if problemDetails != nil {
+		return httpwrapper.NewResponse(int(problemDetails.Status), nil, problemDetails)
+	} else {
+		return httpwrapper.NewResponse(http.StatusNoContent, nil, nil)
+	}
+}
+
+func NfSubscriptionStatusNotifyProcedure(notificationData models.NotificationData) *models.ProblemDetails {
```

Comment:
Should we add a comment to the function for documentation purposes?

Reply:
Comment  is added.